mlir/lib/Dialect/Arith/IR/ArithOps.cpp

```diff
@@ -190,6 +190,14 @@ LogicalResult arith::ConstantOp::verify() {
     return emitOpError(
         "value must be an integer, float, or elements attribute");
   }
+
+  // Intializing scalable vectors with elements attribute is not supported
+  // unless it's a vector splot.
+  auto vecType = dyn_cast<VectorType>(type);
+  auto val = dyn_cast<DenseElementsAttr>(getValue());
+  if ((vecType && val) && vecType.isScalable() && !val.isSplat())
```

Comment:
nit: you can simplify this a bit
```suggestion
  if (vecType && vecType.isScalable() && !isa<SplatElementsAttr>(getValue()))
```

Reply:
btw, have you thought of having a convenience type for scalable vectors? I mean `ScalableVectorType` that is `VectorType` with a classoff function that checks for the scalable bit, similar to how we have `spirv::ScalarType` that allows ints and floats.

Reply:
Interesting idea - I've not thought about it, no. Thanks for the suggestion! Though `isa<ScalableVectorType>(...)` would be longer than `vecType.isScalable()` 🤷🏻 

In general, we've been discussing how to better represent "scalable" vectors in MLIR for a while now. There was also a long discussion on "dynamic" vector type too:
* https://discourse.llvm.org/t/rfc-dynamic-vector-semantics-for-the-mlir-vector-dialect

I know that @dcaballe has been working on an updated class hierarchy for vectors. IIRC, that's not at the top of the priority list ATM, but it would be good to revisit it soon 😅

Reply:
> Interesting idea - I've not thought about it, no. Thanks for the suggestion! Though isa<ScalableVectorType>(...) would be longer than vecType.isScalable() 🤷🏻

only if you have a vector type already. if the starting point is just `Type`, you could do `auto scalableTy = dyn_cast<ScalableVectorType>(myType):`

Reply:
> In general, we've been discussing how to better represent "scalable" vectors in MLIR for a while now. There was also a long discussion on "dynamic" vector type too:

What I suggested above is just a convenience type that doesn't affect the hierarchy in any way, similar to `spirv::ScalarType`.

Reply:
@kuhar I went ahead and landed this to unblock myself elsewhere. I've also tried your suggestion and was positively surprised how straightforward that was:
* https://github.com/llvm/llvm-project/pull/87986

@dcaballe I've prepared #87986 as one point in the discussion on `VectorType` refactor. Mainly so that we have a reference. However, having implemented it I feel that it would indeed be a good stepping stone 😅